addon/components/polaris-button.js

```diff
@@ -322,107 +318,110 @@ export default Component.extend({
-        'isDisabled',
-        'loading',
-        'plain',
-        'monochrome',
-        'size',
-        'fullWidth',
-        'isIconOnly'
-      );
-      let classes = ['Polaris-Button'];
-
-      if (isPresent(externalClasses)) {
-        classes.push(externalClasses);
-      }
-
-      if (primary) {
-        classes.push('Polaris-Button--primary');
-      }
+    'isIconOnly'
+  )
+  get classes() {
+    let {
+      class: externalClasses,
+      primary,
+      outline,
+      destructive,
+      isDisabled,
+      loading,
+      plain,
+      monochrome,
+      size,
+      fullWidth,
+      isIconOnly,
+    } = this;
+    let classes = ['Polaris-Button'];
+
+    if (isPresent(externalClasses)) {
+      classes.push(externalClasses);
+    }
 
-      if (outline) {
-        classes.push('Polaris-Button--outline');
-      }
+    if (primary) {
+      classes.push('Polaris-Button--primary');
+    }
 
-      if (destructive) {
-        classes.push('Polaris-Button--destructive');
-      }
+    if (outline) {
+      classes.push('Polaris-Button--outline');
+    }
 
-      if (isDisabled) {
-        classes.push('Polaris-Button--disabled');
-      }
+    if (destructive) {
+      classes.push('Polaris-Button--destructive');
+    }
 
-      if (loading) {
-        classes.push('Polaris-Button--loading');
-      }
+    if (isDisabled) {
+      classes.push('Polaris-Button--disabled');
+    }
 
-      if (plain) {
-        classes.push('Polaris-Button--plain');
-      }
+    if (loading) {
+      classes.push('Polaris-Button--loading');
+    }
 
-      if (monochrome) {
-        classes.push('Polaris-Button--monochrome');
-      }
+    if (plain) {
+      classes.push('Polaris-Button--plain');
+    }
 
-      if (size && size !== DEFAULT_SIZE) {
-        size = SIZES[size] || null;
-        if (size) {
-          classes.push(`Polaris-Button--size${classify(size)}`);
-        }
-      }
+    if (monochrome) {
+      classes.push('Polaris-Button--monochrome');
+    }
 
-      if (fullWidth) {
-        classes.push('Polaris-Button--fullWidth');
+    if (size && size !== DEFAULT_SIZE) {
+      size = SIZES[size] || null;
+      if (size) {
+        classes.push(`Polaris-Button--size${classify(size)}`);
       }
+    }
 
-      if (isIconOnly) {
-        classes.push('Polaris-Button--iconOnly');
-      }
+    if (fullWidth) {
+      classes.push('Polaris-Button--fullWidth');
+    }
 
-      return classes.join(' ');
+    if (isIconOnly) {
+      classes.push('Polaris-Button--iconOnly');
     }
-  ).readOnly(),
 
-  type: computed('submit', function() {
-    return this.get('submit') === true ? 'submit' : 'button';
-  }).readOnly(),
+    return classes.join(' ');
+  }
+
+  @computed('submit')
+  get type() {
+    return this.submit === true ? 'submit' : 'button';
+  }
 
-  ariaExpandedValue: computed('ariaExpanded', function() {
-    let ariaExpanded = this.get('ariaExpanded');
+  @computed('ariaExpanded')
+  get ariaExpandedValue() {
```

Comment:
this could probably go directly on the template. If an attribute is empty, it won't be rendered anyway.